prow/cmd/deck/main_test.go

```diff
@@ -292,18 +334,37 @@ func TestRerun(t *testing.T) {
 					State: prowapi.PendingState,
 				},
 			})
-			handler := handleRerun(fakeProwJobClient.ProwV1().ProwJobs("prowjobs"), tc.shouldCreateProwjob)
-			req, err := http.NewRequest(http.MethodPost, "/rerun?prowjob=wowsuch", nil)
+			configGetter := func() *config.Config {
+				return &config.Config{
+					ProwConfig: config.ProwConfig{
+						Deck: config.Deck{
+							RerunAuthConfig: config.RerunAuthConfig{
+								AllowAnyone:     tc.allowAnyone,
+								AuthorizedUsers: []string{"hello", "world"},
```

Comment:
I would suggest one of the following:
a) Choosing more obvious names here, like "authorized"
b) even better, moving the AuthorizedUsers list to the template, so its defined right there.

AKA this is more obvious than the current setup:
```golang
			name:                "User not authorized to create prow job",
			login:               "random-dude",
			authorized:          []string{"joe", "susan"},
			allowAnyone:         false,
			rerunCreatesJob:     true,
			shouldCreateProwJob: false,
			httpCode:            http.StatusFound,
			httpMethod:          http.MethodPost,
		},
```